File: storage/models/test_run.go
package models

type TestRun struct {
	Model
	TestID       uint                  `json:"testId"`
	Test         *Test                 `json:"test"`
	AppBinaryID  uint                  `json:"appBinaryId"`
	AppBinary    *AppBinary            `json:"appBinary"`
	StartURL     string                `json:"startUrl"`
	SessionID    string                `json:"sessionId"`
	Parameter    string                `json:"parameter"`
	Protocols    []TestProtocol        `json:"protocols"`
	Log          []TestRunLogEntry     `json:"log"`
	DeviceStatus []TestRunDeviceStatus `json:"deviceStatus"`
}

File: frontend/src/types/test.run.ts
import ITestProtocolData from './test.protocol';
import ITesRunLogEntryData from './test.run.log.entry';
import { TestResultState } from './test.result.state.enum';
import ITestData from './test';
import { IAppBinaryData } from "./app";
import ITesRunDeviceStatusData from "./test.run.device.status";

export default interface ITestRunData {
    id?: number,
    testId: number,
    test: ITestData,
    sessionId: string,
    appBinaryId: number,
    appBinary: IAppBinaryData | null,
    startUrl: string,
    parameter: string,
    testResult: TestResultState,
    protocols: ITestProtocolData[],
    log: ITesRunLogEntryData[],
    deviceStatus: ITesRunDeviceStatusData[],
    createdAt: Date,
    updatedAt: Date,
    deletedAt: Date,
}
